Adding a UseDefaultNamingConvention method to solve multiple DbContexts issue

Azmond opened the pull request:
Adding a UseDefaultNamingConvention method to solve multiple DbContexts with external dependency injection container issue.
And a small addition to the documentation on using the library in this case

#119

roji commented:
@Azmond wouldn't it be better to change the library to default to not doing anything, rather than requiring the user to call UseDefaultNamingConvention?

Out of curiosity... EF Core generally discourages specifynig an external service provider with UseInternalServiceProvider - is there any specific reason you're doing that?
As an aside, note that using an external service provider as you're doing is

roji commented:
Yeah, I'd definitely recommend avoiding UseInternalServiceProvider.

I do think it still makes sense to make this plugin behave well when there's no setting defined - if you want to do that I'd be happy to merge it.

Azmond commented:
I changed the default behavior to do nothing, perhaps this will also rule out other possible errors.
And I corrected the documentation, leaving information about 'AddEntityFrameworkNamingConventions', I think it would be right to mention the possibility of using it.

I understand that isolation is the best. But why is it so highly recommended to avoid UseInternalServiceProvider?

roji commented on EFCore.NamingConventions/NamingConventionsExtensions.cs:
```diff
@@ -102,5 +102,6 @@ public static class NamingConventionsExtensions
             [NotNull] this DbContextOptionsBuilder<TContext> optionsBuilder, CultureInfo culture = null)
             where TContext : DbContext
             => (DbContextOptionsBuilder<TContext>)UseCamelCaseNamingConvention((DbContextOptionsBuilder)optionsBuilder, culture);
+
```
Remove blank line

roji commented on README.md:
````diff
@@ -39,6 +39,16 @@ SELECT c.id, c.full_name
         FROM customers AS c
         WHERE c.full_name = 'John Doe';
 ```
+If you use `UseInternalServiceProvider` to resolve the services from an external dependency injection container, you also need to add the services required for applying naming conventions in Entity Framework Core:
````
Not sure it's a good idea to add this - I'd rather we kept the documentation short and sweet, especially considering that UseInternalServiceProvider isn't recommended EF Core practice.

Azmond replied:
I think it would be correct to briefly mention the AddEntityFrameworkNamingConventions method. And also mention that it is not recommended to use it, as is done in the ef core documentation.

In my case, when using UseInternalServiceProvider and not knowing about AddEntityFrameworkNamingConventions, I initially thought that the library was broken.

I will return the documentation to its original form and leave it up to you, but still believe that any documentation is better than none.

roji reviewed (commented):
Thanks @Azmond, can you please add a regression test as well?

Azmond commented:
Okay, I'll add tests in the coming days as time permits :)

roji commented:
@Azmond apologies this took so long to get around... I'm going over old issues and PR, and I've just submitted #243, which rebases this on latest main with some cleanup. This will be released as part of 8.0.0.

Thanks for your contribution!
